src/zc/queue/_queue.py

```diff
@@ -235,32 +235,23 @@ def __iter__(self):
                 yield i
 
     def __getitem__(self, index):
+        # not efficient, but quick and easy
         if isinstance(index, slice):
-            start, stop, stride = index.indices(len(self))
-            res = []
-            stride_ct = 1
-            for ix, v in enumerate(self):
-                if ix >= stop:
-                    break
-                if ix < start:
-                    continue
-                stride_ct -= 1
-                if stride_ct == 0:
-                    res.append(v)
-                    stride_ct = stride
-            return res
+            # We do this the easy way instead of attempting to handle
+            # negative strides ourself
+            return list(self)[index]
```

Comment:
Oh, wait, there's the docs. I skipped right over that sentence:

> Unlike regular slicing, islice() does not support negative values for start, stop, or step.

So it's actually pretty limited.